packages/lit-html/src/lib/lit-html.ts

```diff
@@ -888,6 +917,58 @@ export class EventPart extends AttributePart {
   }
 }
 
+export class SpreadPart {
+  __value: unknown;
+  __directive?: Directive;
+
+  constructor(public __element: Element) {}
+
+  __setValue(value: unknown): void {
+    const directive = (value as DirectiveResult)._$litDirective$;
+    if (directive !== undefined) {
+      if (this.__directive?.constructor !== directive) {
+        this.__directive = new directive(this);
+        // TODO (justinfagnani): clear?
+      }
+      value = this.__directive.update(this, (value as DirectiveResult).values);
+    }
+    if (value === nothing) {
+      // ?
```

Comment:
Should maybe remove the previous value's bound attributes (if value was of type `AttributesTemplate`)? Should probably do that regardless, if the type of the previous value didn't match new value. This is the tricky part.

Reply:
N/A since removing attr templates